dev/sops.nix

```diff
@@ -0,0 +1,55 @@
+{
+  perSystem =
+    { pkgs, ... }:
+    {
+      packages.sops-check =
+        pkgs.runCommand "sops-check"
+          {
+            buildInputs = with pkgs; [
+              git
+              nix
+              python3.pkgs.deploykit
+              python3.pkgs.invoke
+              sops
+              yq-go
+            ];
+            files = pkgs.lib.fileset.toSource {
+              root = ../.;
+              fileset = pkgs.lib.fileset.unions [
+                (pkgs.lib.fileset.fromSource (pkgs.lib.sources.sourceFilesBySuffices ../. [ ".yaml" ]))
+                ../modules/shared/known-hosts.nix
+                ../sops.json
+                ../sops.nix
+                ../tasks.py
+                ../users
+              ];
+            };
+          }
+          # https://github.com/numtide/treefmt-nix/blob/82bf32e541b30080d94e46af13d46da0708609ea/module-options.nix#L245
+          ''
+            set -e
+            PRJ=$TMP/project
+            cp -r $files $PRJ
+            chmod -R a+w $PRJ
+            cd $PRJ
+            export HOME=$TMPDIR
+            cat > $HOME/.gitconfig <<EOF
+            [user]
+              name = Nix
+              email = nix@localhost
+            [init]
+              defaultBranch = main
+            EOF
+            git init --quiet
+            git add .
+            git commit -m init --quiet
+            export LANG=${if pkgs.stdenv.isDarwin then "en_US.UTF-8" else "C.UTF-8"}
+            export LC_ALL=${if pkgs.stdenv.isDarwin then "en_US.UTF-8" else "C.UTF-8"}
+            export NIX_STATE_DIR=$TMPDIR/state NIX_STORE_DIR=$TMPDIR/store
+            inv update-sops-files || true
+            git status --short
+            git --no-pager diff --exit-code
```

Comment:
This shouldn't be able to actually change any key because `inv update-sops-files` is missing a sops key to actually update the encryption.

Reply:
Yes, update-sops-files won't work. Updated to separate the steps, it'll error if .sops.yaml has a diff and it'll error if any of any of the yaml secrets need to be updated.